hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChunkBufferImplWithByteBuffer.java

```diff
@@ -163,4 +175,12 @@ public String toString() {
     return getClass().getSimpleName() + ":limit=" + buffer.limit()
         + "@" + Integer.toHexString(hashCode());
   }
+
+  @Override
+  protected void finalize() throws Throwable {
```

Comment:
@ChenSammi , Let's don't override `finalize()`.  It will greatly reduce the performance; see HDDS-9295.

Reply:
I see the performance impact. The thing is these MappedByteBuffer are wrapped into ChunkBuffer, then into ByteString, there is not easy to find a good timing other than finalize() and trace back to MappedByteBuffers to release and ummap them.

Reply:
`ChunkBuffer` is an `AutoCloseable`, so can we add cleanup of the `MappedByteBuffer` in `ChunkBuffer#close()`?

Reply:
@ChenSammi , as @adoroszlai mentioned, the buffer should be released in the `close()` method.

BTW, HDDS-7188 ( #3730 ) may be able to fix the problem.

Reply:
ChunkBuffer#close() has been tried, but it's not get called when the buffer is reclaimed.

Reply:
`ChunkBuffer#close()` must be called after use.  If not, it seems a bug.

Reply:
A ChunkBuffer is converted as one ByteString(data can be copied, or zero copied, depending on unsafe conversion is enabled or not). Multiple ByteString data for a readChunk request could be concatenated as one whole ByteString or a list of ByteString, depending on request options.  The whole response with data is passed to the GPRC call which is async. I could be wrong, but I didn't find any GRPC notification or callback to tell a response has been sent out successfully so that the buffer associated with the response can be safely released, when last time I investigated if it's feasible to adopt the buffer pool in datanode for data read.  

When the response is finally sent out, the buffer will be auto released by GC some time later. GC will no call ChunkBuffer#close(), but it will call finalize().

Reply:
I'm trying to use WeakReference solution. Will update the patch later.

Reply:
Please see LeakDetector (HDDS-9528) @ChenSammi

Reply:
GRPC is known to have some limits in the outbound:
 1. It doesn't provide any callbacks or events to notify it successfully sends a response. (unlike when sending a request, you can listen to the response observer).
 2. Even if the buffer being sent is a direct one (or mapped buffer), grpc will not do zero-copy when sending it as a response. It will do another copy to re-frame the original buffer. The copy is done through a reused heap buffer. 

This PR may be very out of scope (of this PR discussion), but GRPC is not the right tool for the job. And we will keep making unnatural efforts to cope with it (like we're doing).